tests/integration/tests/test_version_upgrades.py

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from typing import List
+
+import pytest
+from test_util import config, harness, util
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.mark.node_count(3)
+@pytest.mark.no_setup()
+@pytest.mark.skipif(
+    config.VERSION_UPGRADE_CHANNELS is None, reason="No upgrade channels configured"
```

Comment:
Maybe skip if there aren't any  upgrade channels

```suggestion
    not config.VERSION_UPGRADE_CHANNELS, reason="No upgrade channels configured"
```